File: 進階物件導向專示範/例外處理系統登入示範/src/LoginSystem/LoginSystem.java
package LoginSystem;

public class LoginSystem {
	private final static String ACCOUNT = "123";
	private final static String PASSWORD = "567";
	public static void checkUserAvailable(String account , String password)
	throws AccountNotFoundException,PasswordNotFoundException,PairInfoNotFoundException
	{
		boolean accountError = checkAccountError(account);
		boolean passwordError = checkPasswordError(password);
		if( accountError && passwordError)
			throw new PairInfoNotFoundException();
		if(accountError)
			throw new AccountNotFoundException();
		if(passwordError)
			throw new PasswordNotFoundException();
	}
	public static boolean checkAccountError(String account){
		return !account.equals(ACCOUNT);
	}
	public static boolean checkPasswordError(String password){
		return !password.equals(PASSWORD);
	}
}

File: 一轉示範/例外處理系統登入 python示範/LoginSystem.py
class AccountNotFoundException (Exception): pass
class PasswordNotFoundException (Exception): pass
class PairInfoNotFoundException  (Exception): pass

class LoginSystem:
    ACCOUNT = '000'
    PASSWORD = '567'

    def logIn(self, account, password):
        if account != LoginSystem.ACCOUNT and password != LoginSystem.PASSWORD:
            raise PairInfoNotFoundException()
        if account != LoginSystem.ACCOUNT:
            raise AccountNotFoundException()
        if password != LoginSystem.PASSWORD:
            raise PasswordNotFoundException()
        
       